include/ui_components/surface_cache_helper.hpp

```
@@ -0,0 +1,86 @@
+                                 uint64_t drawHash,
+                                 const SkRect& bounds,
+                                 std::function<void(SkCanvas*)> drawFunc) {
+    // Try to get cached surface
+    sk_sp<SkSurface> cachedSurface = UICacheManager::instance().getCachedSurface(drawHash);
+    if (cachedSurface) {
+      canvas->drawImage(cachedSurface->makeImageSnapshot(), bounds.x(), bounds.y());
+      return true;
+    }
+
+    // Create new surface for caching
+    SkImageInfo info = SkImageInfo::Make(bounds.width(), bounds.height(), kRGBA_8888_SkColorType, kOpaque_SkAlphaType);
```

Clang (14.0.6)

[warning] variable 'info' is not initialized

(cppcoreguidelines-init-variables)